src/dataflow-types/src/types.rs

```diff
@@ -299,30 +299,35 @@ impl Arbitrary for SourceInstanceDesc<CollectionMetadata> {
 
     fn arbitrary_with(_: Self::Parameters) -> Self::Strategy {
         (
+            any::<SourceDesc>(),
             any::<SourceInstanceArguments>(),
             any::<CollectionMetadata>(),
         )
-            .prop_map(|(arguments, storage_metadata)| SourceInstanceDesc {
-                description: crate::sources::any_source_desc_stub(),
-                arguments,
-                storage_metadata,
-            })
+            .prop_map(
+                |(description, arguments, storage_metadata)| SourceInstanceDesc {
+                    description,
+                    arguments,
+                    storage_metadata,
+                },
+            )
             .boxed()
     }
 }
 
 impl RustType<ProtoSourceInstanceDesc> for SourceInstanceDesc<CollectionMetadata> {
     fn into_proto(&self) -> ProtoSourceInstanceDesc {
         ProtoSourceInstanceDesc {
-            description: serde_json::to_string(&self.description).unwrap(),
+            description: Some(self.description.into_proto()),
             arguments: Some(self.arguments.into_proto()),
             storage_metadata: Some(self.storage_metadata.into_proto()),
         }
     }
 
     fn from_proto(proto: ProtoSourceInstanceDesc) -> Result<Self, TryFromProtoError> {
         Ok(SourceInstanceDesc {
-            description: serde_json::from_str(&proto.description)?,
+            description: proto
+                .description
+                .into_rust_if_some("ProtoSourceInstanceDesc::description")?,
```

Comment:
@guswynn is there a way to change the implementation of `into_rust_if_some` so the field argument can be automatically inferred?

For example, here it is `ProtoSourceInstanceDesc::description`, but maybe one can figure it out by reflecting on `self`?

Reply:
there isnt really any reflection in rust, only https://doc.rust-lang.org/std/any/fn.type_name.html which doesnt render things very well

Reply:
I was hoping to get `ProtoSourceInstanceDesc::description` from the call stack somehow.